ext/bz2/bz2.c

```diff
@@ -367,15 +367,15 @@ static PHP_FUNCTION(bzopen)
 	}
 
 	if (mode_len != 1 || (mode[0] != 'r' && mode[0] != 'w')) {
-		php_error_docref(NULL, E_WARNING, "'%s' is not a valid mode for bzopen(). Only 'w' and 'r' are supported.", mode);
-		RETURN_FALSE;
+		zend_value_error("'%s' is not a valid mode for bzopen(). Only 'w' and 'r' are supported.", mode);
+		return;
 	}
 
 	/* If it's not a resource its a string containing the filename to open */
 	if (Z_TYPE_P(file) == IS_STRING) {
 		if (Z_STRLEN_P(file) == 0) {
-			php_error_docref(NULL, E_WARNING, "filename cannot be empty");
-			RETURN_FALSE;
+			zend_value_error("filename cannot be empty");
+			return;
```

Comment:
Do we have any precedent for this? This is generally fine, but I am wondering if it might make sense to keep all the path related error-conditions as warnings (because the main "failed to open file" error will stay that way, and one could see this as a special case of an illegal path).

We also have the CHECK_ZVAL_NULL_PATH check below, which unfortunately doesn't even throw a warning right now, but probably should.

Reply:
I don't think so? I also didn't consider this as a special case of the "failed to open file" error but I do agree in that case it makes more sense to revert it, or can I drop this check and let the stream part raise the warning?

I'll the fix the NULL PATH in a separate PR

Reply:
Seems this has been changed into a type error in the mean time.